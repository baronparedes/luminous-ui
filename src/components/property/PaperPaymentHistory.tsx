import React from 'react';

import {PrintPaper} from '../@print-papers/PaperPdf';
import PaymentHistory from '../@print-papers/PaymentHistory';

type Props = React.ComponentProps<typeof PaymentHistory>;

const PaperPaymentHistory = React.forwardRef<HTMLDivElement, Props>(
  ({...soaProps}, ref) => {
    return (
      <div className="d-none">
        <PrintPaper ref={ref}>
          <PaymentHistory {...soaProps} />
        </PrintPaper>
      </div>
    );
  }
);

export default PaperPaymentHistory;
