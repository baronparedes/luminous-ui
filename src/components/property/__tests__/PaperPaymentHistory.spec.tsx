import {render} from '@testing-library/react';

import PaperPaymentHistory from '../PaperPaymentHistory';

describe('PaperPaymentHistory', () => {
  const props: React.ComponentProps<typeof PaperPaymentHistory> = {
    availablePeriods: ['JAN', 'FEB'],
    year: 2022,
    property: {
      id: 1,
      code: 'G-111',
      address: 'Pasig City Philippines',
      floorArea: 33.1,
      status: 'active',
    },
    paymentHistory: [
      {
        amount: 1000,
        code: 'Charge 1',
        collectedBy: 'Jhon Doe',
        orNumber: 'OR#1',
        paymentType: 'cash',
        transactionPeriod: '2022-01-01',
        createdAt: '2022-02-01',
      },
      {
        amount: 1000,
        code: 'Charge 2',
        collectedBy: 'Jhon Doe',
        orNumber: 'OR#2',
        paymentType: 'check',
        transactionPeriod: '2022-02-01',
        checkIssuingBank: 'Bank of America',
        checkNumber: 'BA-123-123',
        checkPostingDate: '2022-02-28',
        createdAt: '2022-02-01',
      },
    ],
  };

  it('should render', () => {
    const {asFragment} = render(<PaperPaymentHistory {...props} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
