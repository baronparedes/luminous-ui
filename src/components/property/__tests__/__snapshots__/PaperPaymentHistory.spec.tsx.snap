// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaperPaymentHistory should render 1`] = `
<DocumentFragment>
  <div
    class="d-none"
  >
    <div>
      <div
        class="sc-bdfBQB dxnWgn"
      >
        <div
          class="sc-bdfBQB dxnWgn"
        >
          <div
            class="text-center"
          >
            <small>
              <h4
                class="brand"
              >
                Luminous
              </h4>
            </small>
            <h4>
              PAYMENT HISTORY
            </h4>
          </div>
        </div>
        <div
          class="sc-bdfBQB dxnWgn pt-3"
        >
          <div
            class="sc-gsTEea bjzFuM"
          >
            <strong>
              Unit Number: 
            </strong>
            G-111
          </div>
          <small>
            <div
              class="sc-gsTEea bjzFuM"
            >
              <strong>
                Address: 
              </strong>
              Pasig City Philippines
            </div>
          </small>
          <small>
            <div
              class="sc-gsTEea bjzFuM"
            >
              <strong>
                Floor area: 
              </strong>
              33.1
            </div>
          </small>
          <hr />
        </div>
        <div
          class="sc-bdfBQB dxnWgn pt-3"
        >
          <div
            class="sc-gsTEea bjzFuM"
          >
            <strong>
              Payment History for: 
            </strong>
            2022
          </div>
          <hr />
        </div>
        <div
          class="sc-bdfBQB dxnWgn"
        >
          <div
            class="list-group"
          >
            <div
              class="list-group-item"
            >
              <div
                class="container"
              >
                <div>
                  <h5>
                    DATE
                  </h5>
                </div>
              </div>
            </div>
            <div
              class="list-group-item"
            >
              <div
                class="container"
              >
                <div>
                  <h5>
                    DATE
                  </h5>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
